spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/TableWriter.scala

```diff
@@ -204,7 +204,13 @@ class TableWriter[T] private (
       queryExecutor.waitForCurrentlyExecutingTasks()
 
       if (!queryExecutor.successful)
-        throw new IOException(s"Failed to write statements to $keyspaceName.$tableName.")
+        throw new IOException(
+          s"""Failed to write statements to $keyspaceName.$tableName. The
+             |latest exception was
+             |  ${queryExecutor.latestExeception.map( _.getMessage).getOrElse("Unable to determine")}
```

Comment:
It looks like `_successful` (and `successful`) is redundant. The if could be changed to `latestException.map(...)`